Add Azure cloud provider support document

c3y1huang opened the pull request:
# Describe your changes
The skuba added support for in-tree Azure Cloud Provider.

# Related Issues / Projects

Relates to:
* [Support Azure in-tree cloud provider #1582](https://github.com/SUSE/avant-garde/issues/1582)
* [Skuba support In-tree Azure Cloud Provider #1148](https://github.com/SUSE/skuba/pull/1148)
* [New test-cases for Azure as cloud provider #44](https://github.com/SUSE/caasp-test-cases/pull/44)
* [add documents for azure terraform #804](https://github.com/SUSE/doc-caasp/pull/804)

<!--
# Make sure it's tested
*Technical writers will not always be able to verify the implementation!*

Any PR opened is assumed to have been verified by QA if not designated by comments or labels otherwise. Please make sure you have tested the changes and included any information that a user would require to use the documentation.

# Enable maintainer updates
Please enable maintainer updates so we can push commits into your branch to make collaboration and reviews easier.

# Do not force push your branch
Please avoid force pushing to branches that are subject of pull requests. Force pushing breaks maintainer commits in many cases and is very hard (if not impossible) to untangle for backporting.

# Labels
Please set any (and all) appropriate labels that describe the status of the PR.

| **Label** | **Description** |
| --- | --- |
| P1 | PR should be worked on and merged as soon as possible |
| Blocked | Work can not proceed because other work has not been completed, PR can not be merged (code has not been merged but documentation is ready) |
| On-Hold | Underlying work is completed but the PR should not be merged |
| ReleaseNotes | User interaction is required after the introduction of this change and the change must be mentioned in the release notes |
| v3/v4/v4.x | Which version of the release the PR should be merged into, this can be multiple versions, please set the "Backport" label if it needs to go into a previous release |
| Needs Review | Some details of the PR are known to be incomplete and must be discussed with other engineers before merging (if possible assign reviewers or cc mention in comments), PR can not be merged |
-->

Signed-off-by: Chin-Ya Huang <chin-ya.huang@suse.com>

innobead commented on adoc/admin-storage-vsphere.adoc:
```diff
@@ -1,5 +1,5 @@
 = vSphere Storage
-The vSphere cloud provider can be enabled with {productname} to allow Kubernetes pods to use VMWare vSphere Virtual Machine Disk (VMDK) volumes as persistent storage.
+The vSphere cloud provider can be enabled with {productname} to allow Kubernetes pods to use VMWare vSphere Virtual Machine Disk (VMDK) volumes as data volumes.
```
```suggestion
The vSphere cloud provider can be enabled with {productname} to allow Kubernetes pods to use VMWare vSphere Virtual Machine Disk (VMDK) volumes as persistent volumes.
```

jenting commented on adoc/deployment-bootstrap.adoc:
```diff
@@ -290,6 +290,107 @@ these names will be used by the `AWS` cloud controller manager to reconcile node
 You can use the "private dns" values provided by the {tf} output.
 ====
 
+====== Azure In-tree CPI (VCP)
```
It's not VCP (VMWare Cloud Provider), right?

jenting commented on adoc/admin-storage-azure.adoc:
```diff
@@ -0,0 +1,667 @@
+= Azure Storage
+The Azure cloud provider can be enabled with {productname} to allow Kubernetes pods to use Azure disks or Azure files as data volumes.
```
```suggestion
The Azure cloud provider can be enabled with {productname} to allow {kube} pods to use Azure disks or Azure files as persistent volumes.
```

jenting commented on adoc/admin-storage-azure.adoc:
```diff
@@ -0,0 +1,667 @@
+=== Create Disk
+. Install Azure CLI on local machine.
++
+For installation instructions, refer to link:https://docs.microsoft.com/en-us/cli/azure/install-azure-cli?view=azure-cli-latest[Install the Azure CLI].
```
We have `zypper install azure-cli` ^^

c3y1huang replied:
Link includes instructions how to install with zypper.

jenting commented on adoc/deployment-bootstrap.adoc:
```diff
@@ -290,6 +290,107 @@ these names will be used by the `AWS` cloud controller manager to reconcile node
+
+====
+    {
+        "cloud":"AzurePublicCloud",
```
nit
```suggestion
        "cloud": "AzurePublicCloud",
```

jenting commented on adoc/deployment-bootstrap.adoc:
```diff
@@ -290,6 +290,107 @@ these names will be used by the `AWS` cloud controller manager to reconcile node
+
+. Install Azure CLI on local machine.
++
+For installation instructions, refer to link:https://docs.microsoft.com/en-us/cli/azure/install-azure-cli?view=azure-cli-latest[Install the Azure CLI].
```
We have `zypper install azure-cli` ^^

innobead replied:
cc @cclhsu

c3y1huang replied:
Link includes instructions how to install with zypper.

c3y1huang commented:
This will wait until https://github.com/SUSE/doc-caasp/pull/804 gets merged to update parts relates to cluster provision.

innobead commented:
cc @cclhsu

c3y1huang commented:
rebased.

c3y1huang commented:
rebased and fixed IDREF.

r0ckarong commented:
@c3y1huang Added a commit with some minor integration, please rebase before pushing again.

r0ckarong commented:
@c3y1huang Did anything change since the last update for this feature? Please have another review so I can merge this for the 4.5.2 release. Thank you!

c3y1huang commented:
> @c3y1huang Did anything change since the last update for this feature? Please have another review so I can merge this for the 4.5.2 release. Thank you!

Nothing had changed since the last update. This one is ready for review and was just waiting for https://github.com/SUSE/doc-caasp/pull/878#issuecomment-647927137.

c3y1huang commented:
rebased.
